src/AddressRegistry.Projections.Wms/AddressDetail/AddressDetailExtensions.cs

```diff
@@ -8,107 +8,50 @@ namespace AddressRegistry.Projections.Wms.AddressDetail
     using System.Threading.Tasks;
     using Microsoft.EntityFrameworkCore;
     using Microsoft.EntityFrameworkCore.Metadata.Internal;
+    using NetTopologySuite.Geometries;
+    using NetTopologySuite.IO;
 
     public static class AddressDetailExtensions
     {
+        private static string? STAsText(this Point point)
+            => point != null ? WKTWriter.ToPoint(point.Coordinate) : null;
+
         private static async Task<AddressDetailItem> FindAddressDetail(
             this WmsContext context,
             Guid addressId,
-            CancellationToken ct)
+            CancellationToken ct,
+            bool allowRemovedAddress = false)
         {
-            var address = await context
-                .Get<AddressDetailItem>()
-                .FindAsync(addressId, cancellationToken: ct);
+            // because of bulk inserts we needs to handle this a bit different
+            // NOTE: We cannot depend on SQL computed columns when facing with bulk insert that needs to perform queries.
+            // In case of an entry exists in the db but isn't available in local view. It also performs better than Load()
+            await context.AddressDetail.FindAsync(addressId, cancellationToken: ct);
+
+            // 'Local' in our case the source of truth for dealing with bulk operations
+            var address = context.AddressDetail.Local.Single(i => i.AddressId == addressId);
```

Comment:
You're right I'll adjust 
        <summary>
             Finds an entity with the given primary key values. If an entity with the given primary key values
             is being tracked by the context, then it is returned immediately without making a request to the
             database. Otherwise, a query is made to the database for an entity with the given primary key values
             and this entity, if found, is attached to the context and returned. If no entity is found, then
             null is returned.
         </summary>
        public virtual ValueTask<TEntity> FindAsync([CanBeNull] object[] keyValues, CancellationToken cancellationToken)